ecommerce/user/utils.py

```diff
@@ -0,0 +1,113 @@
+from app.database import get_db
+from data.model import Data
+from user.model import Users
+from auth.schemas import UserOut
+from user.password import hash_password
+from sqlmodel import select
+from auth.json_token import SECRET_KEY, ALGORITHM
+
+from jose import jwt, JWTError
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+
+
+router = APIRouter()
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl='login',
+                                     scheme_name="JWT")
+
+
+async def get_current_user(token: str = Depends(oauth2_scheme), database=Depends(get_db)):
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Invalid Credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
+        email: str = payload.get("sub")
+        if email is None:
+            raise credentials_exception
+    except JWTError:
+        raise credentials_exception
+    user = read_user(email, database)
+    if user is None:
+        raise credentials_exception
+    return user
+
+
+@router.post("/Data/create/", tags=['Data'], status_code=201,
+          response_description="Item have been created successfully")
+async def create_data(first_name, last_name, database=Depends(get_db), user: Users = Depends(get_current_user)):
+    database.add(Data(first_name=first_name, last_name=last_name, user_id=user.id))
+    database.commit()
+
+
+def create_user(username, email, password, database):
+    user = Users(username=username, email=email, password=hash_password(password))
+    database.add(user)
+    database.commit()
+    return UserOut(id=user.id, email=user.email)
+
+
+@router.get(f"/Data/read_one/", tags=['Data'])
+async def read(id, database=Depends(get_db), user: Users = Depends(get_current_user)):

```

Comment:
```suggestion
@router.get(f"/data/{data_id}/", tags=['data'])
async def get_data(id, database=Depends(get_db), user: Users = Depends(get_current_user)):
```
REST API pracitces, single object from collection of resources should have path like: `/resource_plural_name/object_id`, for example: `/users/1`.

Also, allow retrieving single object by id or some identifier, since retrieving only first object from database
does not seem useful.